src/main/java/org/jboss/tm/usertx/TransactionListenerRegistry.java

```diff
@@ -0,0 +1,15 @@
+package org.jboss.tm.usertx;
+
+import javax.transaction.NotSupportedException;
+import javax.transaction.Transaction;
+
+public interface TransactionListenerRegistry {
+    /**
+     * @param transaction the transaction that the caller is interested in receiving events for (must not be null)
+     * @param listener an object that will be invoked when events of type {@link org.jboss.tm.usertx.TransactionEvent}
+     *                 occur
+     * @throws NotSupportedException if transaction does not support the listener
+     * feature (or the TSR resource map key is in use
+     */
+    void addListener (Transaction transaction, TransactionListener listener) throws NotSupportedException;
```

Comment:
I chose NotSupportedException because we really do only support transaction types that implement com.arjuna.ats.jta.transaction.Transaction since I need to get the resource map from our private implementation of java.transaction.Transaction.

But now that we are only producing events when the TSR resource map is available I *could* go back to accessing the resource map via the TSR  and drop NotSupportedException. But if, in the future, we do need to support callbacks for events when the TSR is not available (after suspend() for example) then I would need to add the throws NotSupportedException back in again and that would change the interface. What do you think?